### PR TITLE
Fixed updating of /etc/hosts during installation

### PR DESCRIPTION
**Pull request by mchf**
[bsc#1039532](https://bugzilla.suse.com/show_bug.cgi?id=1039532)

### CODE REVIEW

File: src/modules/Host.rb [hunk 1]
```diff
@@ -289,14 +292,22 @@ def StaticIPs
...
+    # It is expected to be used during installation only. If user configures static
+    # ips during installation and do not assign them particular hostname, then such
+    # ips are configuret to resolve to the system wide hostname (see Hostname module,
+    # /etc/HOSTNAME)
+    #
+    # Originally implemented as a fix for bnc#664929, later extended for bnc#1039532
     def ResolveHostnameToStaticIPs
-      static_ips = StaticIPs()
+      # reject those static ips which have particular hostnames already configured
+      static_ips = StaticIPs().reject { |sip| @hosts.include_ip?(sip) }
```
Comment (jreidinger) [inline src/modules/Host.rb]: so looks like variable name is no longer descriptive as it is not static_ips, but static ips without hostname?

File: package/yast2-network.changes [hunk 2]
```diff
@@ -1,4 +1,13 @@
 -------------------------------------------------------------------
+Thu May 25 09:27:09 UTC 2017 - mfilka@suse.com
+
+- bnc1039532
```
Comment (jreidinger) [inline package/yast2-network.changes]: missing `#`

File: src/modules/Host.rb (hunk 1 above)
Comment (mchf) [inline src/modules/Host.rb]: hostname has never been taken into account here

File: package/yast2-network.changes (hunk 2 above)
Comment (mchf) [inline package/yast2-network.changes]: fixed ;-)

File: package/yast2-network.changes [hunk 3]
```diff
@@ -1,4 +1,13 @@
 -------------------------------------------------------------------
+Thu May 25 09:27:09 UTC 2017 - mfilka@suse.com
+
+- bnc#1039532
+  - do not write hostname into /etc/hosts duplicitly for one IP
+  - configure static IPs to resolve to system wide hostname during
+    installation
```
Comment (mvidner) [inline package/yast2-network.changes]: But this was the case already before this fix wasn't it? We still call `ResolveHostnameToStaticIPs`.

**Comment by mvidner** [comment]
From this PR I don't understand what the problem is. Maybe test cases would help?

File: package/yast2-network.changes (hunk 3 above)
Comment (mchf) [inline package/yast2-network.changes]: no, before the hostname was assigned to every static ip - even that which was configured separately.

Comment (mchf) [inline package/yast2-network.changes]: e.g. you could assign ```hostname1``` to ip 1.1.1.1 when editing particular device and ```hostname2``` on Hostname/DNS tab and the result was ```1.1.1.1 hostname1 hostname2```. After this PR it is ```1.1.1.1 hostname1```

**Comment by mchf** [comment]
Referenced issue was partly fixed by previous https://github.com/yast/yast-network/pull/516. Code changes done here are just improvements.

For example before, when hostname was defined via edit device dialog, the /etc/hosts' line looked like
```<ip> <hostname> <hostname>```. Now it is ```<ip> <hostname>```. Other details are described above.

File: package/yast2-network.changes (hunk 3 above)
Comment (mchf) [inline package/yast2-network.changes]: ah I finally see ... I forgot to write half of sentence as I already wrote it 2 or 3 times ;-)

**Comment by mvidner** [comment]
@mchf I have a spaghetti feeling about this. Please add a test for the fix.

**Comment by mchf** [comment]
Thanks for the reviews.

**Pull request merged**